sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ManagedFederatedIdentityCredential.cs

```diff
@@ -0,0 +1,111 @@
+
+        /// <summary>
+        ///   Gets the set of additionally allowed tenants.
+        /// </summary>
+        public IList<string> AdditionallyAllowedTenants { get; }
+        /// <summary>
+        /// Creates an instance of the ManagedFederatedIdentityCredential with a synchronous callback that provides a signed client assertion to authenticate against Microsoft Entra ID.
+        /// </summary>
+        /// <param name="tenantId">The Microsoft Entra tenant (directory) ID of the service principal.</param>
+        /// <param name="clientId">The client (application) ID of the service principal.</param>
+        /// <param name="managedIdentityClientId">The managed identity which has been configured as a Federated Identity Credential (FIC).</param>
+        /// <param name="federatedAudience">
+        ///     The audience for the federated credential, specific to the cloud.  Valid values are:
+        ///       <list type="bullet">
+        ///         <item>
+        ///           <term>api://AzureADTokenExchange</term>
+        ///           <description>Entra ID Global cloud</description>
+        ///         </item>
+        ///         <item>
+        ///           <term>api://AzureADTokenExchangeUSGov</term>
+        ///           <description>Entra ID US Government</description>
+        ///         </item>
+        ///         <item>
+        ///           <term>api://AzureADTokenExchangeChina</term>
+        ///           <description>Entra ID China operated by 21Vianet</description>
+        ///         </item>
+        ///       </list>
+        /// </param>
+        /// <param name="additionallyAllowedTenants">The set of </param>
+        public ManagedFederatedIdentityCredential(string tenantId, string clientId, string managedIdentityClientId, string federatedAudience, IEnumerable<string> additionallyAllowedTenants = default)
+        {
+            ClientAssertionCredentialOptions clientAssertionOptions = null;
+
+            if (additionallyAllowedTenants != null)
+            {
+                clientAssertionOptions = new();
+
+                foreach (var tenant in additionallyAllowedTenants)
+                {
+                    clientAssertionOptions.AdditionallyAllowedTenants.Add(tenant);
+                }
+
+                AdditionallyAllowedTenants = clientAssertionOptions.AdditionallyAllowedTenants;
+            }
+            else
+            {
+                AdditionallyAllowedTenants = new List<string>();
+            }
+
+            _managedIdentityCredential = new ManagedIdentityCredential(managedIdentityClientId);
+            _clientAssertionCredential = new ClientAssertionCredential(
+                tenantId,
+                clientId,
+                async token =>
+                    (await _managedIdentityCredential
+                        .GetTokenAsync(new TokenRequestContext([$"{federatedAudience}/.default"]))
```

Comment:
do we need the `/.default` suffix here?

Reply:
@scottaddie - this is another case of following the article verbatim.   Do we need it?

Reply:
UPDATE: After talking to Bogdan more about this, we decided it's best to keep using the .default suffix. The suffix is an implementation detail. MSI services like to get the audience *without* the suffix, while eSTS likes to get the audience *with* the suffix. By continuing to use the suffix, customers don't have to think about whether it's needed or not.